File: 算法/Java/第1章 算法概述/1.3 基础算法/ChickenRabbit.java
/**
 * （鸡兔同笼）
 * 上有三十五头，下有九十四足，问鸡兔各几何？
 */
public class ChickenRabbit {
	public static void main(String[] args) {
		count(35, 94);
	}
	
	/**
	 * @brief  鸡兔同笼
	 * @param  head: 头数
	 * @param  foot: 脚数
	 * @retval None
	 */
	public static void count(int head, int foot) {
	    for(int chicken = 0; chicken <= head; chicken++) {
	        int rabbit = head - chicken;
	        if(chicken*2 + rabbit*4 == foot) {
	            System.out.println(String.format("鸡：%2d\t兔：%2d", chicken, rabbit));
	        }
	    }
	}
}

File: 算法/Python/第1章 算法概述/1.3 基础算法/chicken_rabbit.py
"""
    （鸡兔同笼）
    上有三十五头，下有九十四足，问鸡兔各几何？
"""

def count(head, foot):
    """
        鸡兔同笼
        Args:
            head (int): 头数
            foot (int): 脚数
    """
    for chicken in range(head+1):
        rabbit = head - chicken
        if chicken*2 + rabbit*4 == foot:
            print("鸡：%2d\t兔：%2d" % (chicken, rabbit))

def main():
    count(35, 94)

if __name__ == "__main__":
    main()